### PR TITLE
generating ingress DR for VS based routing

### PR DESCRIPTION
**Pull request by shriramsharma**
### Description
What does this change do and why?
Generates the DR which applied to the ingress in order to facilitate VS based cross cluster routing

### CODE REVIEW

File: admiral/pkg/clusters/virtualservice_routing.go [hunk 1]
```diff
@@ -16,6 +16,10 @@ import (
 	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+const (
```
Comment (kpharasi) [inline admiral/pkg/clusters/virtualservice_routing.go:19]: Let's move this here: https://github.com/istio-ecosystem/admiral/blob/master/admiral/pkg/util/constants.go?

File: admiral/pkg/clusters/virtualservice_routing.go [hunk 2]
```diff
@@ -619,3 +620,115 @@ func getMeshHTTPPortForRollout(ports map[string]map[string]uint32) (uint32, erro
...
+			continue
+		}
+
+		for _, drHost := range drHosts {
+			drObj := networkingV1Alpha3.DestinationRule{
+				Host:     drHost,
+				ExportTo: common.GetIngressVSExportToNamespace(),
+				TrafficPolicy: &networkingV1Alpha3.TrafficPolicy{
+					LoadBalancer: &networkingV1Alpha3.LoadBalancerSettings{
+						LbPolicy: &networkingV1Alpha3.LoadBalancerSettings_Simple{
```
Comment (vinay-g) [inline admiral/pkg/clusters/virtualservice_routing.go:679]: We need to add something below, else we will have issues where AZ distribution will be a problem - 
```
localityLbSetting:
        distribute:
        - from: us-west-2/*
          to:
            us-west-2/*: 100
        - from: us-east-2/*
          to:
            us-west-2/*: 100
```

File: admiral/pkg/clusters/virtualservice_routing.go [hunk 3]
```diff
@@ -619,3 +620,115 @@ func getMeshHTTPPortForRollout(ports map[string]map[string]uint32) (uint32, erro
...
+	sourceClusterToDRHosts map[string]map[string]string,
+	sourceIdentity string) error {
+
+	if sourceIdentity == "" {
+		return fmt.Errorf("sourceIdentity is empty")
+	}
+
+	san := fmt.Sprintf("%s%s/%s", common.SpiffePrefix, common.GetSANPrefix(), sourceIdentity)
+
+	for sourceCluster, drHosts := range sourceClusterToDRHosts {
```
Comment (vinay-g) [inline admiral/pkg/clusters/virtualservice_routing.go:664]: Question would it better to create one DR with host like - `"*.sandbox-sandbox-vinayclient2-usw2-qal.svc.cluster.local`

File: admiral/pkg/clusters/virtualservice_routing.go [hunk 4]
```diff
@@ -619,3 +620,115 @@ func getMeshHTTPPortForRollout(ports map[string]map[string]uint32) (uint32, erro
...
+						},
+					},
+					Tls: &networkingV1Alpha3.ClientTLSSettings{
+						SubjectAltNames: []string{san},
+					},
+				},
+			}
+			drName := fmt.Sprintf("%s-routing-dr", drHost)
+
+			newDR := createDestinationRuleSkeleton(drObj, drName, util.IstioSystemNamespace)
```
Comment (vinay-g) [inline admiral/pkg/clusters/virtualservice_routing.go:690]: Last time you mentioned VS was it admiral-sync, checking if istio-system will work here

**Review by vinay-g — CHANGES_REQUESTED** [review]

File: admiral/pkg/clusters/virtualservice_routing.go (hunk 3 above)
Comment (shriramsharma) [reply admiral/pkg/clusters/virtualservice_routing.go:664]: I dont see any reason why we cant do that. ok, let me make this change

File: admiral/pkg/clusters/virtualservice_routing.go (hunk 4 above)
Comment (shriramsharma) [reply admiral/pkg/clusters/virtualservice_routing.go:690]: @vinay-g , I'll test out but I think last time I tested, the DR either had to be in the service's NS or in the root NS

Comment (shriramsharma) [reply admiral/pkg/clusters/virtualservice_routing.go:690]: @vinay-g , I tested this out and the DR has to apply either in service's own NS or the root NS (istio-system)

**Review by vinay-g — APPROVED** [review]

**Pull request merged**